include/faabric/scheduler/FunctionMigrationThread.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+
+namespace faabric::scheduler {
+class FunctionMigrationThread
+{
+  public:
+    // Start a background thread that, every wake up period, will check if there
+    // are migration opportunities for in-flight apps that have opted in to
+    // being checked for migrations.
```

Comment:
This is really a description of the whole class, I'd put the comment on the `class`.